pages/api/webhooks.js

```diff
@@ -73,22 +73,27 @@ const webhookHandler = async (req, res) => {
           case 'customer.subscription.deleted':
             await manageSubscriptionStatusChange(
               event.data.object.id,
+              event.data.object.customer,
               event.type === 'customer.subscription.created'
             );
             break;
           case 'checkout.session.completed':
             const checkoutSession = event.data.object;
             if (checkoutSession.mode === 'subscription') {
               const subscriptionId = checkoutSession.subscription;
-              await manageSubscriptionStatusChange(subscriptionId, true);
+              await manageSubscriptionStatusChange(
+                subscriptionId,
+                checkoutSession.customer,
+                true
+              );
             }
             break;
           default:
             throw new Error('Unhandled relevant event!');
         }
       } catch (error) {
         console.log(error);
-        return res.status(400).send('Webhook handler failed. View logs.');
+        return res.json({ error: 'Webhook handler failed. View logs.' });
```

Comment:
No more error status code?

Reply:
@leerob yah, the problem with 400s is that Stripe will eventually turn off your webhook handler if you don't fix it, and I think that would be worse. Is there any way within vercel to warn devs of function errors more explicitly than just `console.log`?

Reply:
Ah, gotcha. Not directly in Vercel, you'd need to persist logs and set up alerting.